src/compiler/scala/tools/nsc/transform/patmat/MatchTranslation.scala

```diff
@@ -237,8 +237,10 @@ trait MatchTranslation {
     // unlike translateMatch, we type our result before returning it
     def translateTry(caseDefs: List[CaseDef], pt: Type, pos: Position): List[CaseDef] =
       // if they're already simple enough to be handled by the back-end, we're done
-      if (caseDefs forall treeInfo.isCatchCase) caseDefs
-      else {
+      if (caseDefs forall treeInfo.isCatchCase) {
+        analyzeCases(NoSymbol, caseDefs.map(translateCase(freshSym(pos, pureType(ThrowableTpe)), AnyValTpe)), AnyValTpe, Suppression.NoSuppression)
```

Comment:
maybe this should take `-Xno-patmat-analysis` into account?

```scala
Welcome to Scala 2.13.0-20180329-170243-4fe7eb9 (Java HotSpot(TM) 64-Bit Server VM, Java 1.8.0_161).
Type in expressions for evaluation. Or try :help.

scala> 1 match { case (i: Int) => (); case y => ??? }
                                                ^
       warning: unreachable code

scala> try 1 catch { case exn: Exception => 2 ; case iae: IllegalArgumentException => 3 }
                                                                                      ^
       warning: unreachable code
res1: Int = 1

scala> :settings -Xno-patmat-analysis

scala> 1 match { case (i: Int) => (); case y => ??? }

scala> try 1 catch { case exn: Exception => 2 ; case iae: IllegalArgumentException => 3 }
                                                                                      ^
       warning: unreachable code
res3: Int = 1
```

Reply:
also, why `pt = AnyValTpe`? As far as I can tell that argument to `translateCase` only ever winds up being used in `BodyTreeMaker#toString` but it's not obvious what it's for. (And not at all in `analyzeCases`.

Maybe the `pt` arg can just be removed? Unless IntelliJ is lying to me again.

Reply:
I don't disagree, but I was just getting something to compile. There's also a comment about tree duplication on the next line which may or may not apply. I didn't have time this morning to become an expert.